spec/lib/notify_delivery_method_spec.rb

```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+RSpec.describe NotifyDeliveryMethod do
+  describe "#deliver!" do
+    it "calls Notify's send_email endpoint" do
+      headers = { to: "x@y.com", template_id: SecureRandom.uuid }
+      personalisation = { body: "Body content", subject: "A subject line" }
+      message = Mail::Message.new(headers.merge(personalisation))
+
+      client = instance_double("Notifications::Client")
+      allow(Notifications::Client).to receive(:new).and_return(client)
+
+      expect(client).to receive(:send_email)
+        .with(email_address: headers[:to],
+              template_id: headers[:template_id],
+              personalisation: personalisation)
+
+      described_class.new.deliver!(message)
```

Comment:
I think this is our first usage of described_class. It'd be more consistent, for this codebase, to go with `NotifyDeliveryMethod`.

Aside from that though this test looks good 👍